content/docs/rfcs/27/README.md

```diff
@@ -0,0 +1,98 @@
+---
+slug: 27
+title: 27/WAKU2-PEERS
+name: Waku v2 Client Peer Management Recommendations
+status: raw
+editor: Hanno Cornelius <hanno@status.im>
+contributors:
+---
+
+`27/WAKU2-PEERS` describes a recommended minimal set of peer storage and peer management features to be implemented by Waku v2 clients.
+
+In this context, peer _storage_ refers to a client's ability to keep track of discovered or statically-configured peers and their metadata.
+It also deals with matters of peer _persistence_,
+or the ability to store peer data on disk to resume state after a client restart.
+
+Peer _management_ is a closely related concept and refers to the set of actions a client MAY choose to perform based on its knowledge of its connected peers,
+e.g. triggering reconnects/disconnects, keeping certain connections alive, etc.
+
+# Peer store
+
+The peer store SHOULD be an in-memory data structure where information about discovered or configured peers are stored.
+It SHOULD be considered the central authority for peer-related information in a Waku v2 client.
```

Comment:
I wouldn't use the word "central authority" when talking about local knowledge in your own node - makes it seem like it is some kind of external CA or something

Reply:
Yeah, I had a bad conscience about using "central" here. The point is to strongly recommend that clients should avoid distributed peer metadata internally.

Will just change to `main authority`

Reply:
> should avoid distributed peer metadata internally.
Could you elaborate?

Reply:
maybe "main source of truth"? "authority" to me suggests it is external, internal might be "sovereign"

Reply:
> Could you elaborate?

Yeah, simply that peer metadata that might influence our behaviour towards that peer should preferably be kept in one place to avoid making bad decisions. One example is always using the manager when dialing/disconnecting from a peer to keep proper track of sessions in one place. Another example would be if SWAP Accounting determines a peer should be blacklisted, ensuring that `ShouldntConnect` information is visible to the peer manager when any different protocol, say `relay`, tries to connect.